ferry/transform/transform_compute.py

```diff
@@ -135,19 +135,20 @@ def evaluation_statistics_computed(
     def average_by_course(question_tag: str, n_categories: int):
         tagged_ratings = evaluation_ratings[evaluation_ratings["tag"] == question_tag]
         # course_id -> Series[list[int]]
-        rating_by_course = tagged_ratings.groupby("course_id")["rating"]
+        rating_by_course = tagged_ratings.groupby("course_id")[["rating", "question_code"]]
 
-        def average_rating(data: pd.Series) -> float:
+        def average_rating(data: pd.DataFrame) -> float:
             # A course can have multiple questions of the same type. This usually
             # happens when the course is cross-listed between GS and YC
-            weights = [sum(x) for x in zip(*data)]
-            if len(weights) != n_categories:
+            weights = [sum(x) for x in zip(*data["rating"])]
+            question_code = list(data["question_code"])[0]
+            if len(weights) != n_categories and question_code != "DR359":
```

Comment:
```suggestion
            if len(weights) != n_categories and not (data["question_code"] == "DR359").all():
```

Could you try this? Also add a comment about what DR359 is

Reply:
done!